streams/src/main/java/org/apache/kafka/streams/kstream/internals/KStreamImpl.java

```diff
@@ -305,13 +306,23 @@ public void to(final String topic) {
         to(topic, Produced.<K, V>with(null, null, null));
     }
 
-    @SuppressWarnings("unchecked")
     @Override
     public void to(final String topic, final Produced<K, V> produced) {
-        Objects.requireNonNull(topic, "topic can't be null");
+        Objects.requireNonNull(topic, "topic chooser can't be null");
```

Comment:
there is no topic chooser here.

Reply:
@guozhangwang missed this one (just a nit though)

Reply:
Sorry, missed this one.. updated.